Course/Course/Data/CourseRepository.swift

```diff
@@ -245,7 +245,15 @@ public class CourseRepository: CourseRepositoryProtocol {
                 .replacingOccurrences(of: "?lang=\($0.key)", with: "")
             return SubtitleUrl(language: $0.key, url: url)
         }
-            
+        
+        let type: BlockType = {
```

Comment:
@shafqat-muneer Do you think we need to override the block type? I can imagine a situation where we would need to know the original type (e.g. we want to have different icons for .html and .unknown types).
Perhaps we could add changes to LessonType like this:
<img width="691" alt="Screenshot 2024-09-16 at 15 25 09" src="https://github.com/user-attachments/assets/566aa625-393a-465b-a2a8-918d626dde6f">
WDYT?
Thank you

Reply:
@rnr, after a bit of analysis, I agree that it's more appropriate to handle it here. Although we don't have such a case at the moment, there's a chance it could occur in the future.

I've made the update. Thanks for pointing it out!